Commit to add changes made to haproxy.j2 template

k8soneill opened the pull request:
This adds some configuration to the haproxy backend atomic-openshift-api to improve the failover time when a master server goes down. Changes made are 2 small additions to the /etc/haproxy/haproxy.cfg file under the backend block for atomic-openshift-api. timeout check 2s and timeout connect 2s have been added, this means that a health check will time out if no response is received from the server after 2 seconds of being connected and also connections to the backend will timeout if no connection is established in 2 seconds. As these are on a local network 2 seconds should be more than enough time for connections to be established/health checks to respond and this brings the failover time down to around 8 seconds.

This is to rectify issue #51

Commit:
Commit to add changes made to haproxy.j2 template that adds some configuration to the haproxy backend atomic-openshift-api to improve the failover time when a master server goes down. Changes made are 2 small additions to the /etc/haproxy/haproxy.cfg file under the backend block for the previously mentioned backend. timeout check 2s and timeout connect 2s have been added, this means that a health check will time out if no response is received from the server after 2 seconds of being connected and also connections to the backend will timeout if no connection is established in 2 seconds. As these are on a local network 2 seconds should be more than enough time for connections to be established/health checks to respond and this brings the failover time down to around 8 seconds.

stevemul commented:
Looks good to me, have you tested the deployment to ensure it successfully deploys this config and works as you'd expect? It looks like it will, but we need to be certain. Also looks like this addresses #29 as well - didn't realise we already had an issue for this problem.

We also need some tests written that test master failover.

Commit:
Created test to check failover time for master node haproxy backend. Ansible playbook polls API endpoint, bash script runs the playbook and stops the API services on the master node in turn then calculates downtime and outputs a file with the connections/attempts.

Commit:
Modified the test so that it hard reboots the master nodes in order to properly simulate the loss of a node in production environment.

Commit:
Removed obsolete directory that wasn't removed in last commit.

Commit:
Playbooks for tests moved into relevant test folders and comments added to playbooks. Minor changes to absolute path names in bash script too.

Commit:
Added an openshift login line to start of script in order to run oc commands and also added exit code of 1 against exit calls.

k8soneill committed:
Update master-failover-test.sh

Commit:
Amended openshift login to use ADMIN_USERNAME and ADMIN_PASSWORD variables set in pipeline.

Commit:
Merge branch 'bugfix/haproxy_timeout_changes' of github.com:UKCloud/openshift-deployment-ansible into bugfix/haproxy_timeout_changes

Conflicts:
	tests/haproxy/master-failover-test.sh

Commit:
Amended line in openshiftpostdeployment roles main.yml to give admin cluster-admin role as was not working before.

k8soneill committed:
Reverted admin rights command

Issue was that postdeployment code would not run due to failures earlier in the playbook rather than the command itself

k8soneill committed:
Adding fix for service broker for pipeline test

Previous playbook needs to complete in order for my test to complete. Adding this in for pipeline test.

k8soneill committed:
set -e added to deploy script to pickup command failures

k8soneill committed:
upgrade_bastion.yml commented out

upgrade_bastion.yml being removed for now until package dependency issue can be looked into/resolved.

Commit:
added oc project command to test script in order to change to default project for haproxy test.

k8soneill committed:
Updated test

Amended path to ansible log and added some debug value outputs to test behaving as expected.

k8soneill committed:
Update master-failover-test.sh

Commit:
Added change directory to test script and local config/logfile for ansible.

Commit:
Removed become on the poll.yml as pipeline doesn't like it

Commit:
Changed all paths to be absolute to make this work in pipeline.

Commit:
Added another absolute path that was missed

Commit:
Fixed scripts to work over SSH on bastion in pipeline.

k8soneill committed:
Update master-failover-test.sh

k8soneill committed:
Changed oc login command

To ensure it can login without verifying tls

k8soneill committed:
Update master-failover-test.sh

k8soneill committed:
Update master-failover-test.sh

k8soneill committed:
Update master-failover-test.sh

k8soneill committed:
Update master-failover-test.sh

k8soneill committed:
Cleaned up script

Tidied up debugging from script for final test run

k8soneill commented:
Tested in pipeline configuration deploys successfully and test script(also integrated into jenkins pipeline) shows failover times reduced to 9-11 seconds

k8soneill committed:
Merge branch 'master' into bugfix/haproxy_timeout_changes

k8soneill committed:
Added auth URL variable

k8soneill committed:
Update master-failover-test.sh
